internal/provider/kubernetes/controller.go

```diff
@@ -196,7 +207,45 @@ func newGatewayAPIController(mgr manager.Manager, cfg *config.Server, su status.
 	return nil
 }
 
+func (r *gatewayAPIReconciler) enqueueManagedClass(obj client.Object) []reconcile.Request {
```

Comment:
curious why this function Is needed instead of just capturing the gateway class and the envoy proxy associated with it here https://github.com/envoyproxy/gateway/blob/50e0bcb936884195ec1ce9d03bd2503bcbad7c7a/internal/provider/kubernetes/controller.go#L250

Reply:
`cc` and `acceptedGC` are created during a `Reconcile()`. This method creates the reconcile request for the GC that ref's the watched EnvoyProxy. EnvoyProxy does not bind/ref other resources, e.g. parentRef, so the method must list all GCs to see which one references the watched EnvoyProxy. This is similar to listing all HTTPRoutes for a Gateway CRUD to understand which routes are used to calculate listener conditions.